### PR TITLE
Resurrect ocl subscription test

### PR DESCRIPTION
**Pull request by kdaud**

### CODE REVIEW

**Comment by sherrif10** [comment]
Lets create  ticket for this issue.

**Comment by kdaud** [comment]
Check my recent comment on `pull/379`

**Comment by kdaud** [comment]
@sherrif10 when we pass the server's credentials directly as strings the OCL Subscription Module test passes actions.

**Comment by sherrif10** [comment]
> Check my recent comment on `pull/379`

can you add the link to it here.

File: qaframework-bdd-tests/cypress/integration/cucumber/step_definitions/oclsubscriptionmodule/ocl-subscription-module.js [hunk 1]
```diff
@@ -60,13 +53,13 @@ Then('the user should copy the subscription url', () => {
 });
 
 When('the user visit the OCL module page', () => {
-  cy.visit('http://localhost:8080/openmrs/owa/openconceptlab/index.html#/subscription')
+  cy.visit('https://demo.openmrs.org/openmrs/owa/openconceptlab/index.html#/subscription')
```
Comment (sherrif10) [inline qaframework-bdd-tests/cypress/integration/cucumber/step_definitions/oclsubscriptionmodule/ocl-subscription-module.js:56]: Any reason why you are using demo instead of local host.

**Comment by kdaud** [comment]
> can you add the link to it here

https://github.com/openmrs/openmrs-contrib-qaframework/pull/379

**Comment by sherrif10** [comment]
@kdaud  isn't this this test supposed to be segregated. I hope to see only one workflow getting build

**Comment by sherrif10** [comment]
The recent commit @ibacher  shared was to help us run workflows per a given workflow. We don't need to see all workflows getting triggered. So I still don't get it when I see many workflows getting triggered.

**Comment by kdaud** [comment]
The reason why it triggers tests for 2.x is that have added a feature file within the path "qaframework-bdd-tests/src/**/*.*" hence triggering the 2,x tests to run. If the feature file was not added then your expectation would be true. Just in case you want to verify I can temporarily delete out the feature file added in that path and then only OCL Subscription Module test will run.

**Comment by kdaud** [comment]
@sherrif10 what do you think of this PR?

File: qaframework-bdd-tests/cypress/integration/cucumber/step_definitions/oclsubscriptionmodule/ocl-subscription-module.js [hunk 2]
```diff
@@ -25,15 +18,15 @@ Given('the user goes to the openmrs application', () => {
...
+  cy.get("#username").type("openmrs");
+  cy.get("#password").type("Openmrs123");
   cy.get("#login-form form").submit();
   cy.url().should('contain', 'https://openmrs.staging.openconceptlab.org/user/collections/');
 });
 
 When('the user logs into the openmrs Ref App', () => {
-  cy.get("#username").type(OPENMRS_USER_NAME);
-  cy.get("#password").type(OPENMRS_USER_PASSWORD);
+  cy.get("#username").type("admin");
```
Comment (sherrif10) [inline qaframework-bdd-tests/cypress/integration/cucumber/step_definitions/oclsubscriptionmodule/ocl-subscription-module.js:28]: @kdaud   why this change. what happens when its left the way it was initially.

**Comment by sherrif10** [comment]
@kdaud  i can see some un addressed comments.

File: qaframework-bdd-tests/cypress/integration/cucumber/step_definitions/oclsubscriptionmodule/ocl-subscription-module.js (hunk 2 above)
Comment (kdaud) [reply qaframework-bdd-tests/cypress/integration/cucumber/step_definitions/oclsubscriptionmodule/ocl-subscription-module.js:28]: The server credentials fails to be exported hence throwing Cypress error: `cy.type() can not accept an empty string`, on the the other hand as suggested in the PR when the credentials are passed directly as strings the test passes.

**Comment by kdaud** [comment]
> @kdaud i can see some un addressed comments.

Do you mind pointing me to the un addressed comment?

File: qaframework-bdd-tests/cypress/integration/cucumber/step_definitions/oclsubscriptionmodule/ocl-subscription-module.js (hunk 1 above)
Comment (sherrif10) [reply qaframework-bdd-tests/cypress/integration/cucumber/step_definitions/oclsubscriptionmodule/ocl-subscription-module.js]: Any response on the above comment.

File: qaframework-bdd-tests/cypress/integration/cucumber/step_definitions/oclsubscriptionmodule/ocl-subscription-module.js (hunk 2 above)
Comment (sherrif10) [reply qaframework-bdd-tests/cypress/integration/cucumber/step_definitions/oclsubscriptionmodule/ocl-subscription-module.js:28]: > `cy.type() can not accept an empty string`,

You mean even when you  equate that string  to a value, something like String` USER_NAME  = "Admin"`, it still fails ?

File: qaframework-bdd-tests/cypress/integration/cucumber/step_definitions/oclsubscriptionmodule/ocl-subscription-module.js (hunk 1 above)
Comment (kdaud) [reply qaframework-bdd-tests/cypress/integration/cucumber/step_definitions/oclsubscriptionmodule/ocl-subscription-module.js]: Sure I reverted the server back to the local instance that is why it's showing `Outdated` label.

File: qaframework-bdd-tests/cypress/integration/cucumber/step_definitions/oclsubscriptionmodule/ocl-subscription-module.js (hunk 2 above)
Comment (kdaud) [reply qaframework-bdd-tests/cypress/integration/cucumber/step_definitions/oclsubscriptionmodule/ocl-subscription-module.js:28]: The credentials are set in command.js file
```js
export const OCL_USER_NAME = Cypress.env('OCL_USER_NAME');
export const OCL_USER_PASSWORD = Cypress.env('OCL_USER_PASSWORD');
export const OPENMRS_USER_NAME = Cypress.env('OPENMRS_USER_NAME');
export const OPENMRS_USER_PASSWORD = Cypress.env('OPENMRS_USER_PASSWORD');
export const OCL_USER_TOKEN = Cypress.env('OCL_USER_TOKEN'
```
which are then imported as 

```js
import { 
  OCL_USER_NAME, 
  OCL_USER_PASSWORD, 
  OCL_USER_TOKEN,
  OPENMRS_USER_NAME, 
  OPENMRS_USER_PASSWORD
} from '../../../../support/commands';
```
Then passed in their respective fields. However they are all failing to be imported hence throwing the error I shared above.

Comment (sherrif10) [reply qaframework-bdd-tests/cypress/integration/cucumber/step_definitions/oclsubscriptionmodule/ocl-subscription-module.js:28]: Are they exported, Usually react js, js classes need to be exported first for them to be accessed.

Comment (sherrif10) [reply qaframework-bdd-tests/cypress/integration/cucumber/step_definitions/oclsubscriptionmodule/ocl-subscription-module.js:28]: @kdaud  Seems even the export function does not work as expected. Look here https://github.com/openmrs/openmrs-contrib-qaframework/blob/master/qaframework-bdd-tests/cypress/support/commands.js#L10

Comment (kdaud) [reply qaframework-bdd-tests/cypress/integration/cucumber/step_definitions/oclsubscriptionmodule/ocl-subscription-module.js:28]: Do you have a suggestion that can use rather?

Comment (sherrif10) [reply qaframework-bdd-tests/cypress/integration/cucumber/step_definitions/oclsubscriptionmodule/ocl-subscription-module.js:28]: If this really failed with that initial work, then we can go with direct placing of those attribute.

Comment (kdaud) [reply qaframework-bdd-tests/cypress/integration/cucumber/step_definitions/oclsubscriptionmodule/ocl-subscription-module.js:28]: Have not gotten a way of importing them after trying several alternatives I know. If there is one you know then we can use that rather otherwise we can have them passed directly as strings and if we get a way out  then we can do refactor the test.

Comment (sherrif10) [reply qaframework-bdd-tests/cypress/integration/cucumber/step_definitions/oclsubscriptionmodule/ocl-subscription-module.js:28]: > we can have them passed directly as strings and if we get a way out then we can do refactor the test.

I agree because i have gone through it microscopically and am conviced this looks good.

**Review by sherrif10 — APPROVED** [review]
Thanks @kdaud

**Comment by sherrif10** [comment]
Thanks @kdaud  Hopefuly you will be winking on the build of the OCL part.

**Comment by kdaud** [comment]
@sherrif10  I see Green Lights after the build plan. The test is back to life!
Screenshot from 2022-03-28 12-09-46

**Comment by sherrif10** [comment]
ohh yeah man. Awesome.
